### PR TITLE
feat: homepage

### PR DESCRIPTION
**Pull request by bhtibrewal**
Added context, basic components, pages structure, basic routing

pages to check:
- header.jsx
- sidebar.jsx
- contexts
pages to avoid 
- CSS files

### CODE REVIEW

**Review by sri-rishi — APPROVED** [review]
code looks fine to me, feel free to merge

File: src/context/ThemeContext.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const ThemeProvider = ({ children }) => {
+  const preferDarkMode = matchMedia("(prefers-color-scheme: dark)").matches;
+  const localMode = JSON.parse(localStorage.getItem("mode"));
+  const [darkMode, setDarkMode] = useState(localMode ?? preferDarkMode);
+  useEffect(() => {
+    localStorage.setItem("mode", darkMode);
+
+    return () => {
+      localStorage.clear("mode");
```
Comment (anasm11) [inline src/context/ThemeContext.jsx:13]: Is it required to write the statement & return the function here?

**Review by anasm11 — APPROVED** [review]
Icons arent visible in the header. Rest all looks good.

File: src/context/ThemeContext.jsx (hunk 1 above)
Comment (bhtibrewal) [reply src/context/ThemeContext.jsx:13]: I think this is the way to write cleanup functions

**Pull request merged**